### ISSUE TITLE
NaN loss when there are 0s in cooccurrence matrix

### ISSUE THREAD

**Issue by MJafarMashhadi** [erwtokritos/keras-glove#2]
0s in co-occurrence matrix make `y_true = 0` which makes `K.Log(y_true) = Inf`.
It results in the loss and gradients to become `NaN` and in turn makes the embedding layer full of `NaN`s and the model becomes untrainable right after the first epoch.

**Comment by erwtokritos** [comment]
Hello. Thank you for your comment! I understand your point but the fix that you propose is not actually needed. As you can see in https://github.com/erwtokritos/keras-glove/blob/master/app/text_utils.py#L55 I am not computing the 'full' cooccurrence matrix between all possible word pairs, I just keep a cache of all pairs with non-zero values and thus `y_true` (i.e. Xij) cannot be zero. Even if I kept the whole co-occurrence matrix then this wouldn't affect the training process due to the `f` function introduced by the authors which is 0 for Xij = 0 (see page 4 of the paper https://nlp.stanford.edu/pubs/glove.pdf).

**Comment by MJafarMashhadi** [comment]
Oh yeah that's right! I was using only a part of your implementation; I had the full co-occurrence matrix built elsewhere which stored the 0s as well.
